docs/internal/publishing.md

```diff
@@ -2,9 +2,15 @@
 
 Follow these instructions for releasing a new version of Mesop publicly via PyPI (e.g. `pip install mesop`).
 
-## Bump the version
+If you haven't done this before, follow the [first-time setup](#first-time-upload-setup).
 
-Update `mesop/version.py` by incrementing the version number. We follow semver.
+## Update version to RC
+
+Update [`mesop/version.py`](https://github.com/google/mesop/blob/main/mesop/version.py) by incrementing the version number. We follow semver.
+
+You want to first create an RC (release candidate) to ensure that it works.
+
+For example, if the current version is: `0.7.0`, then you should increment the version to `0.8.0rc0` which will create an RC, which is treated as a [pre-release by PyPI](https://packaging.python.org/en/latest/specifications/version-specifiers/#pre-releases).
```

Comment:
[question] I got a bit confused here. I'm also not too familiar with release candidate versioning stuff.

In the example, you say if the version is 0.7.0, then we should increment to version 0.8.0rc0. Is that always true. Would it be possible do a 0.7.1rc0 for a minor patch change or is the rc process only for incremental major/minor versions?

Reply:
I think you can do 0.7.1rc0

Because we are pre v1, we aren't super strict about versioning. Technically in semver 0.7 to 0.8 can contain breaking changes, but in my mind, it could just also be a bunch of new features.

I think we can just stick to bumping the middle version, unless it's a really small release (e.g. 1 bug fix)